overlord/configstate/configcore/hostname.go

```diff
@@ -51,7 +51,7 @@ func validateHostnameSettings(tr config.ConfGetter) error {
 	if hostname == "" {
 		return nil
 	}
-	if !validHostname(hostname) {
+	if len(hostname) > 253 || !validHostname(hostname) {
```

Comment:
nitpick: reading this line makes me think that a valid hostname can have any length, because the two checks are separate.

Nitpicky suggestion: create a `validHostname()` function, which both checks the length and applies the regexp.